nms/app/packages/magmalte/app/views/metrics/LteMetricsData.js

```diff
@@ -0,0 +1,783 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+export type MetricsDetail = {
+  name: string,
+  promQL: string,
+  description: string,
+  category: string,
+  service: string,
+};
+
+export const LteMetricsTable: Array<MetricsDetail> = [
```

Comment:
Do we have a way to programmatically generate these? Or otherwise perhaps do a sync with prometheus to check when metrics are added/removed and this list requires update?

Reply:
This would be the ideal case. We should easily be able to programmatically generate the metric list. What we lack is the descriptions along with it. We should improve the metrics model and provide this as an API from orc8r so that we can display it and it is consistent with the changes we do as we add the metrics.

Reply:
Once this PR is merged, we should follow up with an issue which tracks adding these metrics programmatically.

Reply:
Yeah how did you go about generating these this time? Also I think it would be best to pull out this list of metrics into its own .json file and then load it in where it's needed. Otherwise you're going to have a LOT of data as code

Reply:
It is from the static metrics document which we have. For our next release, we have to have some visibility into metrics. Currently there is no information on metrics outside of the graphs and grafana and partners struggle with the lack of info. So this is a kind of workaround, till we improve the API to return us this entire dataset.

Reply:
Discussed offline but putting here for posterity: Since all metrics are generated using the prometheus client library, we could generate these docs by just finding all of those calls (e.g. `prometheus.GAUGE('name', 'help text', ['labels'])`) and parsing the information out of those. Would require getting the AST and such but should be fairly simple for python and Go.

Alternatively you can parse the information from `<prometheus_server>:9090/metrics` which gives a list of all metric series in the database. This is not 100% accurate however as some metrics are only generated if certain conditions are met, so unless the deployment you get this from has hit all of those cases, you'll miss some metrics.